lib/mongify/translation/sync.rb

```diff
@@ -56,23 +56,22 @@ def setup_sync_table
       # Does the straight copy (of tables)
       def sync_data
         self.copy_tables.each do |t|
-          q = "SELECT t.* FROM #{t.sql_name} t, #{SYNC_HELPER_TABLE} u " +
-            "WHERE t.updated_at > u.last_updated_at AND u.table_name = '#{t.sql_name}'"
-          rows = sql_connection.select_by_query(q)
-          Mongify::Status.publish('copy_data', :size => rows.count, :name => "Syncing #{t.name}", :action => 'add')
-          max_updated_at, max_updated_at_id = Time.new(1970), nil
-          rows.each do |row|
-            row_hash = t.translate(row)
-            updated_at = Time.parse(row['updated_at'])
-            if updated_at > max_updated_at
-              max_updated_at = updated_at
-              max_updated_at_id = row_hash['pre_mongified_id']
+          sql_connection.select_rows(t.sql_name, true) do |rows, page, total_pages|
+            Mongify::Status.publish('copy_data', :size => rows.count, :name => "Syncing #{t.name} (#{page}/#{total_pages})", :action => 'add')
```

Comment:
Line is too long. [142/80]<br>Use the new Ruby 1.9 hash syntax.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.